### PR TITLE
Update Shell Scripting section in CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by rawiriblundell**
Hi,
In this commit, I have updated the Shell Scripting section in CONTRIBUTING.md to provide more emphasis on portability.

I have added extra notes around shellcheck usage.

I've also removed the line:

>Set `set -e -o pipefail` at the top of your script

The so-called "Unofficial Strict Mode" and variants of it are well-intentioned but unfortunately flawed.  See [this post](https://www.reddit.com/r/commandline/comments/g1vsxk/the_first_two_statements_of_your_bash_script/fniifmk/) for a thorough explanation as to why.  You may also like to read the following:

* http://mywiki.wooledge.org/BashFAQ/105
* http://mywiki.wooledge.org/BashFAQ/112
* https://wiki.bash-hackers.org/scripting/obsolete

Cheers!

### CODE REVIEW

**Comment by jonaskluger** [comment]
Hey Rawiri,

thank you for this pull request. We will have a closer look at your patch and discuss internally on how and if we can implement it into the official code base.

Internal Ref: CMK-4299

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -604,35 +604,222 @@ about the error diagnosis below, if it doesn't work.
...
+this up in the near future.
+
+Do note that while Shellcheck is an excellent tool, it's not perfect.
+Sometimes it alerts on code that may actually be desired.  In this scenario,
+you must use a `disable` directive with a comment that justifies your reason
+for this.  Preferably this will be a git commit hash where the reasoning is
+given e.g.
+
+```
+# See commit abcd3456
```
Comment (LarsMichelsen) [inline CONTRIBUTING.md:782]: Better than referring to a commit is to directly add the reason to the comment line(s) above.

Comment (rawiriblundell) [reply CONTRIBUTING.md]: Reworded in the latest commit(s)

**Comment by svenpanne** [comment]
I've merged this PR more or less unchanged after rebasing it (1b5298c9f76ac70110f781028d0dc270c8851a6f), the general advice seems sensible. Perhaps we tweak things a bit in the future, who knows... :wink:  Thanks for the PR!

**Pull request closed without merging**